src/main/java/de/tum/in/www1/jenkins/notifications/SendTestResultsNotificationPostBuildTask.java

```diff
@@ -106,6 +117,25 @@ public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath filePath, @Nonnull
                 .collect(Collectors.toList());
     }
 
+    private List<String> extractLogs(@Nonnull Run<?, ?> run, TaskListener taskListener) {
+        final List<String> logs = new ArrayList<>();
+        final StringWriter stringWriter = new StringWriter();
+
+        try {
```

Comment:
I would suggest to use [try-with-resources](https://www.baeldung.com/java-try-with-resources) for the StringWriter (should be available since Java 1.7)